gammapy/scripts/jupyter.py

```diff
@@ -0,0 +1,157 @@
+
+
+@click.command(name='execute')
+@click.pass_context
+def cli_jupyter_execute(ctx):
+    """Execute jupyter notebook."""
+
+    for path in build_nblist(ctx):
+        run_notebook(path)
+
+
+@click.command(name='test')
+@click.pass_context
+def cli_jupyter_test(ctx):
+    """Check if Jupyter notebook is broken."""
+
+    for path in build_nblist(ctx):
+        run_notebook(path)
+        rawnb = read_notebook(path)
+
+        if rawnb:
+            report = ""
+            passed = True
+            log.info("   ... Testing {}".format(str(path)))
+            for cell in rawnb.cells:
+                if 'outputs' in cell.keys():
+                    for output in cell['outputs']:
+                        if output['output_type'] == 'error':
+                            passed = False
+                            traceitems = ["--TRACEBACK: "]
+                            for o in output['traceback']:
+                                traceitems.append("{}".format(o))
+                            traceback = "\n".join(traceitems)
+                            infos = "\n\n{} in cell [{}]\n\n" \
+                                    "--SOURCE CODE: \n{}\n\n".format(
+                                        output['ename'],
+                                        cell['execution_count'],
+                                        cell['source']
+                                    )
+                            report = report + infos + traceback
+            if passed:
+                log.info("   ... {} Passed".format(str(path)))
+            else:
+                log.info("   ... {} Failed".format(str(path)))
+                log.info(report)
+
+
+def tag_magics(cellcode):
+    """Comment magic commands when formatting cells."""
+
+    lines = cellcode.splitlines(False)
+    for line in lines:
+        if line.startswith("%") or line.startswith("!"):
+            magic_line = "###-MAGIC COMMAND-" + line
+            yield magic_line
+        else:
+            yield line
+
+
+def build_nblist(ctx):
+    """Fill list of notebooks in a the given folder."""
+
+    if ctx.obj['file']:
+        yield Path(ctx.obj['file'])
+    else:
+        for f in Path(ctx.obj['fold']).iterdir():
+            if f.name.endswith('.ipynb'):
+                yield f
+
+
+def run_notebook(path):
+    """Execute a Jupyter notebook."""
+
+    try:
+        t = time.time()
+        subprocess.call(
+            "jupyter nbconvert "
+            "--allow-errors "
+            "--ExecutePreprocessor.timeout=None "
+            "--ExecutePreprocessor.kernel_name=python3 "
+            "--to notebook "
+            "--inplace "
+            "--execute '{}'".format(path),
+            shell=True)
+        t = (time.time() - t) / 60
+        log.info('Executing duration: {:.2f} mn'.format(t))
+    except Exception as ex:
+        log.error('Error executing file {}'.format(str(path)))
+        log.error(ex)
+
+
+def read_notebook(path):
```

Comment:
Is this `read_notebook` function useful?

You call it three times. In one case you do handle the "none" error case, in the other two you don't and will fail with a cryptic AttributeError in that case I think.

So this exception is for corrupt Jupyter notebook files? I think it would be OK to just let `nbformat.read` error out on those and halt, because in practice that doesn't occur, and if it does, it's clear that there's a corrupt file, no?

If you agree, you could just have the `nbformat.read` call in every current caller of `read_notebook` and delete this helper function.